UnityProject/Assets/Scripts/Core/Random13.cs

```diff
@@ -58,11 +58,44 @@ public static bool Prob()
 		{
 			if (Instance == null)
 			{
-				throw new NotImplementedException("Random13 was never created and stored anywhere.");
+				throw new Exception("Random13 was never created and stored anywhere.");
 			}
 			Instance.ProgressIndex();
-			if (Instance.RandomNumber % 2 == 0) return true;
-			return false;
+			return Instance.RandomNumber % 2 == 0;
+		}
+
+		/// <summary>
+		/// Consistent random numbers for things like damage.
+		/// </summary>
+		public static float RandomNumberWave(float baseNumber)
+		{
+			if (Instance == null)
```

Comment:
since you are doing thischeck and raise the exception in all usages, it would be better to move it to the property declaration

Reply:
wdym?

Reply:
```cs
public static Random13 Instance => instance ?? throw new Exception("you didn't blah blah") : instance;

private static Random13 instance;
```

And you would assign the private field in the constructor. That would make it so any time someone tries to access the public property, it will throw if it is null and you don't need to check it in every method.

Reply:
ah

Reply:
I messed up the C# syntax, long time without writing any dotnet. But you get the gist of what I tried to do.